mitodl/micromasters
# Zendesk prepopulate name/email

Partial fix for #1262. This pull request sets the name and email address for the logged-in user in the [ZenDesk web widget](https://developer.zendesk.com/embeddables/docs/widget/api). A significant amount of the work in this pull request is in serializing the User object, and testing that serialization. I was running into import errors when I tried to keep the tests in the `profiles` module, so I ended up moving the serialization machinery, as well as the tests for it, into the `micromasters` module.

Note that this results in some duplication on the `SETTINGS` object. We can clean that up in a later pull request -- changing all the existing references would make this pull request even larger.
#### How should this be manually tested?

Open the ZenDesk widget while logged out, and verify that it is not populated with information. Then log in, and open the ZenDesk widget again, and verify that the "name" and "email" fields are pre-populated. You can also view the source of the page, and verify that the user object is getting serialized to the `SETTINGS` object in Javascript.
